exporter/loggingexporter/config.go

```diff
@@ -15,28 +15,87 @@
 package loggingexporter // import "go.opentelemetry.io/collector/exporter/loggingexporter"
 
 import (
+	"fmt"
+
 	"go.uber.org/zap/zapcore"
 
 	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/config/configtelemetry"
+	"go.opentelemetry.io/collector/confmap"
 )
 
 // Config defines configuration for logging exporter.
 type Config struct {
 	config.ExporterSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
 
 	// LogLevel defines log level of the logging exporter; options are debug, info, warn, error.
+	// Deprecated: Use `Verbosity` instead.
 	LogLevel zapcore.Level `mapstructure:"loglevel"`
 
+	// Verbosity defines the logging exporter verbosity.
+	Verbosity configtelemetry.Level `mapstructure:"verbosity"`
+
 	// SamplingInitial defines how many samples are initially logged during each second.
 	SamplingInitial int `mapstructure:"sampling_initial"`
 
 	// SamplingThereafter defines the sampling rate after the initial samples are logged.
 	SamplingThereafter int `mapstructure:"sampling_thereafter"`
+
+	// warnLogLevel is set on unmarshaling to warn users about `loglevel` usage.
+	warnLogLevel bool
 }
 
 var _ config.Exporter = (*Config)(nil)
+var _ confmap.Unmarshaler = (*Config)(nil)
+
+func mapLevel(level zapcore.Level) (configtelemetry.Level, error) {
+	switch level {
+	case zapcore.DebugLevel:
+		return configtelemetry.LevelDetailed, nil
+	case zapcore.InfoLevel:
+		return configtelemetry.LevelNormal, nil
+	case zapcore.WarnLevel, zapcore.ErrorLevel,
+		zapcore.DPanicLevel, zapcore.PanicLevel, zapcore.FatalLevel:
+		// Anything above info is mapped to 'basic' level.
+		return configtelemetry.LevelBasic, nil
+	default:
+		return configtelemetry.LevelNone, fmt.Errorf("log level %q is not supported", level)
+	}
+}
+
+func (cfg *Config) Unmarshal(conf *confmap.Conf) error {
+	if conf.IsSet("loglevel") && conf.IsSet("verbosity") {
+		return fmt.Errorf("'loglevel' and 'verbosity' are incompatible. Use only 'verbosity' instead")
+	}
+
+	if err := conf.Unmarshal(cfg, confmap.WithErrorUnused()); err != nil {
+		return err
+	}
+
+	if conf.IsSet("loglevel") {
+		verbosity, err := mapLevel(cfg.LogLevel)
+		if err != nil {
+			return fmt.Errorf("failed to map 'loglevel': %w", err)
+		}
+
+		// 'verbosity' is unset but 'loglevel' is set.
+		// Override default verbosity.
+		cfg.Verbosity = verbosity
+		cfg.warnLogLevel = true
+	}
+
+	return nil
+}
 
 // Validate checks if the exporter configuration is valid
 func (cfg *Config) Validate() error {
+	switch cfg.Verbosity {
+	case configtelemetry.LevelBasic, configtelemetry.LevelNormal, configtelemetry.LevelDetailed:
+		// These are supported. Do nothing.
+	default:
+		// configtelemetry.LevelNone and future values are not supported.
+		return fmt.Errorf("verbosity level %q is not supported", cfg.Verbosity)
+	}
```

Comment:
nit: somehow this is difficult to read, I really like more the style "if !ok { return }" ways of checking the conditions. If you have an idea how to do that please do, otherwise it is fine.

Reply:
Done with a set (map to empty struct)